[자동차 경주] 원영진 과제 제출합니다.

AI-WonYJ opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **신규 기능**
  - 사용자 입력에 대한 검증(이름 길이 제한 및 숫자 입력 확인)이 강화되었습니다.
  - 각 라운드의 진행과 점수 집계, 그리고 최종 승자 결정 기능이 새롭게 도입되어 게임 플레이 경험이 개선되었습니다.
  - 게임 시작 시 자동차 이름과 시도 횟수를 입력받는 인터랙티브한 흐름이 추가되었습니다.
  - README 문서에 게임의 입력 및 출력 기능에 대한 자세한 설명이 추가되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AI-WonYJ committed:
feat: racingcar 초기 모델

racingcar 최초 모델 생성성

AI-WonYJ committed:
style: trailing-whitespace 제거

62번줄, 65번줄 trailing-whitespace

AI-WonYJ committed:
style: PEP8 Style로 코드 변경

PEP8 Style로 코드 변경

AI-WonYJ committed:
style: PEP8 Style로 코드 변경

PEP8 Style로 코드 변경

AI-WonYJ committed:
fix: is_number()함수 기능 오류 해결

int형인지 검사해야하지만, str형인지로 검사하는 오류

swthewhite commented on src/racingcar/main.py:
```diff
@@ -1,10 +1,71 @@
+import random
+
+
+def is_number(Data):
```
pameter는 소문자로 시작하는 것이 권장됩니다.
Editer의 `리팩토링-Rename` 기능으로 수정해보시죠.

swthewhite commented on src/racingcar/main.py:
```diff
@@ -1,10 +1,71 @@
+        if len(Data[i]) > 5:
+            raise ValueError("이름은 5자 이하만 가능합니다.")
+        Data[i] = [Data[i], 0]
+    return Data
```
빈 이름에 대한 검증이 힘들 것 같네요.

swthewhite commented on src/racingcar/main.py:
```diff
@@ -1,10 +1,71 @@
+
+def try_input():
+    print("시도할 횟수는 몇 회인가요?")
+    n = input()
```
코딩테스트가 아닌 경우에는 변수명을 더 명확히 작성해주는 것이 좋습니다.

swthewhite commented on src/racingcar/main.py:
```diff
@@ -1,10 +1,71 @@
+    print("시도할 횟수는 몇 회인가요?")
+    n = input()
+    is_number(n)
+    return int(n)
```
시도 횟수에 대한 추가 검증이 있다면 좋을 것 같네요.

swthewhite commented on src/racingcar/main.py:
```diff
@@ -1,10 +1,71 @@
+    for i in range(n):
+        name_list = play_1set_of_game(name_list)
+        for i in range(len(name_list)):
+            print("{0} : {1}".format(name_list[i][0], "-" * name_list[i][1]))
```
함수로 바꾸는 게 더 가독성이 좋을 것 같습니다.

swthewhite commented on src/racingcar/main.py:
```diff
@@ -1,10 +1,71 @@
+def play_1set_of_game(Data):
+    for i in range(len(Data)):
+        value = random.randint(0, 10)
+        if value >= 4:
```
숫자상수는, 매직넘버상수로 바꾸고 전방선언 해주는 것이 좋습니다.

swthewhite reviewed (approved):
전반적으로 잘 하셨어요.
요구사항에 맞춰 조금만 더 코드를 개선해봅시당~

AI-WonYJ committed:
refactor: 게임 실행 결과 출력 기능 분리

게임 실행 결과가 출력되는 기능을 main()으로부터 'print_game_play_result()'라는 함수로 분리

AI-WonYJ committed:
refactor: Change parameter to lowercase

대문자로 시작하는 파라미터를 소문자로 시작하도록 변경

AI-WonYJ committed:
fix: Add validation for empty names

빈 이름에 대한 검증 추가
중복되는 이름에 대한 검증 추가가

AI-WonYJ committed:
refactor: Write variable names more clearly

변수명을 더 정확히 작성 (n -> try_number)

AI-WonYJ committed:
refactor: Write variable names more clearly

변수명을 더 정확히 작성 (n -> try_number)

AI-WonYJ committed:
fix: Add validation for number of attempts

시도 횟수에 대한 검증 추가 (양의 정수가 아니면 ValueError 발생)

AI-WonYJ committed:
refactor: Change to magic number constant & PEP8 Style

숫자 상수를 매직넘버 상수로 바꾸고, 전방선언
PEP8 Style로 코드 변경

AI-WonYJ committed:
refactor: Change to PEP8 Style

아래 스타일 오류 수정
src/racingcar/main.py:81:65: E231 missing whitespace after ','
src/racingcar/main.py:81:80: E501 line too long (82 > 79 characters)
src/racingcar/main.py:93:11: W292 no newline at end of file

AI-WonYJ committed:
refactor: Change the validation method for entering of attempts

시도 횟수 입력에 대한 검증 방법 변경
validate_input()에 instance를 str로 가져온 다음 is_number()로 검증 후 양의
정수인지 확인하는 방법 -> try_input()에서 is_number()로 검증 후
validate_input()에 instance를 int형으로 보내어 코드 가독성을 높임

AI-WonYJ committed:
refactor: Separater car name input function

자동차 이름을 입력받는 함수를 분리
전체적인 코드 가독성 높임

AI-WonYJ committed:
docs: Write functional documentation
